### PR TITLE
INTYGFV-15210: Implementation of QuestionSjukskrivningsperiod. Added CertificateDataConfigDateRange and testFixtures for DateRange value & config.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Comment by ChrilleSigerhed** [comment]
image

File: skl/ag114/src/test/java/se/inera/intyg/common/ag114/v1/validator/InternalDraftValidatorImplTest.java [hunk 1]
```diff
@@ -201,6 +202,8 @@ public void testFailsWhenInvalidSjukskrivningsperiod() throws ScenarioNotFoundEx
         Assert.assertEquals(SJUKSKRIVNINGSPERIOD_SVAR_JSON_ID_7_2 + ".tom", error.getField());
         assertEquals(ValidationMessageType.INCORRECT_COMBINATION, error.getType());
         assertEquals(COMMON_VALIDATION_DATE_PERIOD_INVALID_ORDER, error.getMessage());
+        assertEquals(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_ID, error.getQuestionId());
+
```
Comment (mhornfeldt) [inline skl/ag114/src/test/java/se/inera/intyg/common/ag114/v1/validator/InternalDraftValidatorImplTest.java:206]: This extra spacing could be removed

**Review by mhornfeldt — APPROVED** [review]
Nice! :)

File: skl/ag114/src/test/java/se/inera/intyg/common/ag114/v1/validator/InternalDraftValidatorImplTest.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply skl/ag114/src/test/java/se/inera/intyg/common/ag114/v1/validator/InternalDraftValidatorImplTest.java:206]: Fixed!

**Comment by ChrilleSigerhed** [comment]
Added changes to expression to better cater for frontend validation. Result:
image

File: skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsperiod.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    .id(SJUKSKRIVNINGSGRAD_PERIOD_JSON_ID)
+                    .text(textProvider.get(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_TEXT_ID))
+                    .fromLabel(textProvider.get(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_LABEL_FROM_ID))
+                    .toLabel(textProvider.get(SJUKSKRIVNINGSGRAD_PERIOD_SVAR_LABEL_TOM_ID))
+                    .build()
+            )
+            .value(
+                CertificateDataValueDateRange.builder()
+                    .id(SJUKSKRIVNINGSGRAD_PERIOD_JSON_ID)
+                    .from(sjukskrivningsPeriod != null && sjukskrivningsPeriod.getFrom() != null ? sjukskrivningsPeriod.fromAsLocalDate()
```
Comment (mww-sopra-steria) [inline skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsperiod.java:59]: Redundant check. sjukskrivningsPeriod.fromAsLocalDate() handles if from is null, so you could simplify the statement by removing && sjukskrivningsPeriod.getFrom() != null.

Same from getTom().

**Review by mww-sopra-steria — APPROVED** [review]
Nice! 

A minor suggestion to simplify a statement. If agreed, change before merging.

File: skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsperiod.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply skl/ag114/src/main/java/se/inera/intyg/common/ag114/v1/model/converter/certificate/question/QuestionSjukskrivningsperiod.java]: Good suggesstion! Fixed!

**Pull request merged**